packages/jest-circus/src/legacy-code-todo-rewrite/jestAdapterInit.ts

```diff
@@ -80,6 +80,9 @@ export const initialize = async ({
     xtest: globals.it.skip,
   };
 
+  globalsObject.test.deadline = () =>
```

Comment:
is this needed?

Reply:
This is where end users would read the deadline from; it makes `test.deadline()` available from within a test.

If you want to expose that at all, it would be useful for my "pass it along with RPC calls" case.

Reply:
Since we use `expect.withinDealine()` wouldn't it make more sense as `expect.deadline()`?

---

I wonder if it makes even more sense on `TestContext`, i.e. `this.deadline()`. Not 100% sure, tho.

Reply:
I've moved this to `expect.deadline()`, which happens to make the implementation waaay less spread out. Coupled with moving the storage to the runner state directly (as `currentlyRunningChildDeadline`), we now don't have to fiddle with event handlers, don't have to compute the stuff there, etc.

For the terminology, I actually quite like `test.deadline`. When is the test deadline? What's the test's deadline? What's the expected deadline, but we're not enforcing it, it's just for your information. What do I expect the deadline to be? How do I expect something completes within a deadline? The test should do something within a deadline.